internal/resolution/entitysources/catalogdsource.go

```diff
@@ -110,9 +110,10 @@ func getEntities(ctx context.Context, client client.Client) (input.EntityList, e
 				if catalogScopedEntryName == bundle.Name {
 					channelValue, _ := json.Marshal(property.Channel{ChannelName: ch.Name, Priority: 0})
 					props[property.TypeChannel] = string(channelValue)
-					// TODO(jmprusi): Add the proper PropertyType for this
-					replacesValue, _ := json.Marshal(replacesProperty{Replaces: b.Replaces})
-					props["olm.replaces"] = string(replacesValue)
+					replacesValue, _ := json.Marshal(replacesProperty{
+						Replaces: fmt.Sprintf("%s-%s%s%s", bundle.Spec.Catalog.Name, b.Replaces, bundle.Spec.Package, ch.Name),
```

Comment:
So the idea is that: in the entity properties we have formated replaces (vs raw data from catalogd objects) and don't have to format it in `internal/resolution/variablesources/installed_package.go`: we just filter entity source by id of the currently installed entity.

However I'm not entirely sure whether this is what we want to do. This limits replaces to a specific catalog and channel.

Reply:
In OLM V0, which I believe is be the primary usecase for replace/skips/skipRange based upgrades, we supported cross channel updates. I suspect that we'll want to support cross channel upgrades here as well.

Reply:
Yeah, and for [semver upgrade support](https://docs.google.com/document/d/1KIhV8oqr22Bp8lf370Aej2n-3uW_xGcTptONHHPRVYA/edit) we also want:

* The bundle resolved for an operator during an upgrade can come from a different channel or catalog than the currently installed bundle. (i.e. successors can come from any channel and any catalog)
* The bundle resolved for an operator during an upgrade cannot come from a different package than the currently installed bundle.

So I'm looking into other options to solve this.

Reply:
I think it may be less restricting to have channel information be a new constraint rather than included into the replaces field keeping cross channel updates in mind, similar to what the upgrade support brief mentions.